kotlin/android/app/src/main/java/dev/firezone/android/tunnel/TunnelService.kt

```diff
@@ -198,6 +203,11 @@ class TunnelService : VpnService() {
     private fun shutdown() {
         Log.d(TAG, "shutdown")
 
+        val connectivityManager =
+            getSystemService(ConnectivityManager::class.java) as ConnectivityManager
+        connectivityManager.unregisterNetworkCallback(networkCallback!!)
```

Comment:
```suggestion
        networkCallback?.let {
        
            val connectivityManager =
                getSystemService(ConnectivityManager::class.java) as ConnectivityManager
            connectivityManager.unregisterNetworkCallback(networkCallback)
            networkCallback = null
        }
```

I think I assumed that `shutdown()` would be safe to call idempotently.

Reply:
Or better yet move this to a helper function, `stopNetworkMonitoring(connlibSessionPtr)` like the Apple client.

Reply:
why would we want to call `shutdown` multiple times in a row?

Reply:
I feel like `shutdown` should only be called when the tunnel is no longer used.

Reply:
We don't, but there's an edge case if onDisconnect and the user disconnect at the same time.

The system can shut us down too but I don't think that calls disconnect.

Just trying to avoid crashing in our only cleanup handler is all.

Reply:
Ah, I see, then we probably want to change `disconnect` to not use `connlibSessionPtr!!`, I can do that on this PR

Reply:
changed it here: https://github.com/firezone/firezone/pull/4163/commits/0ce223beb2c93c72b96ae2f9cc8a75fecd271a92 lmk if I should roll the change on `disconnect` back :)